examples/subplots_axes_and_figures/broken_axis.py

```diff
@@ -5,61 +5,50 @@
-    0.109, 0.195, 0.050, 0.074, 0.079, 0.155, 0.020, 0.010, 0.061, 0.008])
+np.random.seed(19680801)
 
+pts = np.random.rand(30)*.2
 # Now let's make two outlier points which are far away from everything.
 pts[[3, 14]] += .8
 
 # If we were to simply plot pts, we'd lose most of the interesting
 # details due to the outliers. So let's 'break' or 'cut-out' the y-axis
-# into two portions - use the top (ax) for the outliers, and the bottom
+# into two portions - use the top (ax1) for the outliers, and the bottom
 # (ax2) for the details of the majority of our data
-f, (ax, ax2) = plt.subplots(2, 1, sharex=True)
+fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True)
+fig.subplots_adjust(hspace=0.05)  # adjust space between axes
 
 # plot the same data on both axes
-ax.plot(pts)
+ax1.plot(pts)
 ax2.plot(pts)
 
 # zoom-in / limit the view to different portions of the data
-ax.set_ylim(.78, 1.)  # outliers only
+ax1.set_ylim(.78, 1.)  # outliers only
 ax2.set_ylim(0, .22)  # most of the data
 
 # hide the spines between ax and ax2
-ax.spines['bottom'].set_visible(False)
+ax1.spines['bottom'].set_visible(False)
 ax2.spines['top'].set_visible(False)
-ax.xaxis.tick_top()
-ax.tick_params(labeltop=False)  # don't put tick labels at the top
+ax1.xaxis.tick_top()
+ax1.tick_params(labeltop=False)  # don't put tick labels at the top
 ax2.xaxis.tick_bottom()
 
-# This looks pretty good, and was fairly painless, but you can get that
-# cut-out diagonal lines look with just a bit more work. The important
-# thing to know here is that in axes coordinates, which are always
-# between 0-1, spine endpoints are at these locations (0, 0), (0, 1),
-# (1, 0), and (1, 1).  Thus, we just need to put the diagonals in the
-# appropriate corners of each of our axes, and so long as we use the
-# right transform and disable clipping.
-
-d = .015  # how big to make the diagonal lines in axes coordinates
-# arguments to pass to plot, just so we don't keep repeating them
-kwargs = dict(transform=ax.transAxes, color='k', clip_on=False)
-ax.plot((-d, +d), (-d, +d), **kwargs)        # top-left diagonal
-ax.plot((1 - d, 1 + d), (-d, +d), **kwargs)  # top-right diagonal
+# Now, let's turn towards the cut-out slanted lines.
+# We create line objects in axes coordinates, in which (0,0), (0,1),
+# (1,0), and (1,1) are the four corners of the axes.
+# The slanted lines themselves are markers at those locations, such that the
+# lines keep their angle and position, independent of the axes size or scale
+# Finally, we need to disable clipping.
 
-kwargs.update(transform=ax2.transAxes)  # switch to the bottom axes
-ax2.plot((-d, +d), (1 - d, 1 + d), **kwargs)  # bottom-left diagonal
-ax2.plot((1 - d, 1 + d), (1 - d, 1 + d), **kwargs)  # bottom-right diagonal
+d = .5  # proportion of vertical to horizontal extent of the slanted line
+kwargs = dict(marker=[(-1, -d), (1, d)], markersize=12,
+              linestyle="none", color='k', mec='k', mew=1, clip_on=False)
```

Comment:
perhaps use mew=rcParams["axes.linewidth"] here? 1 looks a bit thick (the rc value is 0.8).

Reply:
Unfortunately the default of 0.8 looks not so nice when not being straight (for the default dpi and pixel output)

![image](https://user-images.githubusercontent.com/23121882/70803558-a18e5580-1db4-11ea-8cd8-bce06b380de1.png)

Hence I chose to keep 1.

![image](https://user-images.githubusercontent.com/23121882/70803597-b539bc00-1db4-11ea-8c07-38534612a784.png)

Reply:
I actually think 0.8 looks better than 1 (hence the suggestion) but am fine with your choice too.